basis/pure/basisComputeLib.sml

```diff
@@ -16,7 +16,11 @@ val add_basis_compset = computeLib.extend_compset
      mlintTheory.maxSmall_DEC_def,
      mlintTheory.padLen_DEC_eq,
      mlintTheory.toChars_def,
-     mlintTheory.toString_def
+     mlintTheory.toString_def,
+     mlnumTheory.toString_def,
+     mlnumTheory.num_toString_def,
+     mlnumTheory.fromString_def,
+     mlnumTheory.fromString_unsafe_def
```

Comment:
The lack of these definitions was the main culprit of the whole `"not terminated by EMPTYSTRING"` issue